website/docs/opengl/c1/3-hello-texture.md

```diff
@@ -0,0 +1,388 @@
+
+out vec4 out_color;
+
+void main()
+{
+    // This will allow us to see the texture coordinates in action!
+    out_color = vec4(frag_texCoords.x, frag_texCoords.y, 0, 1.0);
+}
+```
+
+Now we just need to assign the correct layout for `aTextureCoord`. You should add these lines below our first `VertexAttribPointer` call:
+
+```c#
+const uint texCoordLoc = 1;
+_gl.EnableVertexAttribArray(texCoordLoc);
+_gl.VertexAttribPointer(texCoordLoc, 2, VertexAttribPointerType.Float, false, 5 * sizeof(float), (void*)(3 * sizeof(float)));
+```
+
+Pay attention to the last parameter, the pointer! It represents the size of bytes that should be jumped at the start of
+each stride. It means GL will add 3 indexes after stride and pick the consecutive 2 values.
+if you do everything right, you will see this result!
+
+![Quad with UVs](../../../images/opengl/chapter1/leasson3/quad-with-uvs.png)
+
+But we aren't there just yet. We can't see any texture in our quad!
+What you are seeing is the texture coordinates we passed displayed as a color!
+
+What does this mean? The colors that you see are the texture coordinate values of that pixel. As a way to demonstrate it, we're converting
+the numeric values to color values to see what we are doing. In the shader, the values are being read like this:
+
+![UV values](../../../images/opengl/chapter1/leasson3/quad-with-uvs-and-numbers.png)
+
+The neat part is that the fragment shader already interpolates the coordinates! So we don't need to do that ourselves, and can just send a value per
+vertex. It saves a lot of work on our end!
+
+## Importing images as textures
+Now for the fun part: rendering an image!
+
+To start this section, first download
+[silk.png](https://github.com/dotnet/Silk.NET/blob/477154b99564aac9cf82426754857103fb88d2ea/examples/CSharp/OpenGL%20Tutorials/Tutorial%201.3%20-%20Textures/silk.png).
+We will use this image in this tutorial!
+
+In most cases with OpenGL, textures are uploaded as a sequence of bytes, this is usually done with a layout of 4 bytes per pixel (Red, Green, Blue, and Alpha).
+The hard part is: we can't just upload the bytes of a .png or .jpg file! These formats have a lot of unnecessary data, like headers and most importantly
+compression, which the GPU generally cannot understand.
+
+To load an image file as a byte array, we will first need a external library. In our case, we will use StbImageSharp.
+
+First of all, install the library using NuGet. To do so, run the following command:
+```
+dotnet add package StbImageSharp
+```
+
+After installing, add the following reference on the top of your code file:
+```c#
+using StbImageSharp;
+```
+
+At the top of your class, declare a `uint` to store the ID of the OpenGL texture object:
+```c#
+private static uint _texture;
+```
+
+Returning to your `OnLoad` method, add this line at the end of the method to create, and bind the texture.
+That `ActiveTexture` call is telling OpenGL that we are wanting to use the first texture slot.
```

Comment:
You haven't explained what a texture slot (texture unit) actually is.

Reply:
I prefer if we called them texture units and not slots, as that's how they're referred to by the official Khronos docs

Reply:
There is now an info tag with this explanation